geode-core/src/main/java/org/apache/geode/internal/net/SCAdvancedSocketCreator.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.net;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+import javax.net.SocketFactory;
+
+import org.apache.geode.GemFireConfigException;
+import org.apache.geode.SystemConnectException;
+import org.apache.geode.distributed.internal.tcpserver.AdvancedSocketCreatorImpl;
+import org.apache.geode.distributed.internal.tcpserver.ConnectionWatcher;
+import org.apache.geode.distributed.internal.tcpserver.HostAndPort;
+
+class SCAdvancedSocketCreator extends AdvancedSocketCreatorImpl {
+  final SocketCreator coreSocketCreator;
+
+  protected SCAdvancedSocketCreator(SocketCreator socketCreator) {
+    super(socketCreator);
+    coreSocketCreator = socketCreator;
+  }
+
+  @Override
+  public void handshakeIfSocketIsSSL(Socket socket, int timeout) throws IOException {
+    coreSocketCreator.handshakeIfSocketIsSSL(socket, timeout);
+  }
+
+  @Override
+  public Socket connect(HostAndPort addr, int timeout,
+      ConnectionWatcher optionalWatcher, boolean allowClientSocketFactory, int socketBufferSize,
+      boolean useSSL) throws IOException {
+
+    coreSocketCreator.printConfig();
+
+    if (!useSSL) {
+      return super.connect(addr, timeout, optionalWatcher, allowClientSocketFactory,
+          socketBufferSize,
+          useSSL);
+    }
+
+    // create an SSL connection
+
+    Socket socket;
+    InetSocketAddress sockaddr = addr.getSocketInetAddress();
```

Comment:
`getSocketInetAddress()` will try to resolve the hostname stored in `addr`…

Reply:
Yes, that's what the LocatorAddress method was supposed to do.  I renamed LocatorAddress to HostAndPort since it can hold arbitrary addresses, not just locator addresses.  For your PR you can bypass getSocketInetAddress and just get the hostname since that's all you need for the SNIServerName parameter.